pkg/storage/unified/search/bleve_test.go

```diff
@@ -1186,6 +1186,53 @@ func TestIndexUpdate(t *testing.T) {
 	require.Equal(t, int64(5), searchTitle(t, idx, "gen_2", 10, ns).TotalHits)
 }
 
+func TestConcurrentIndexUpdateAndBuildIndex(t *testing.T) {
```

Comment:
iiuc, after we build the first index, this test does:

1. The first call to UpdateIndex() starts the updater (it will run in its goroutine for 5 secs).
2. Builds a new index, which closes the first index and shuts down it's updater.
3. Calls UpdateIndex() on the closed index - which should fail since the index is closed.